core/tests/webdriverio_utils/ExplorationEditorMainTab.js

```diff
@@ -439,8 +440,13 @@ var ExplorationEditorMainTab = function() {
 
     await editOutcomeDestDropdownOptions.selectByVisibleText(targetOption);
 
+    // 'End' is one of the keys name for setValue function, so to insert
+    // 'End' as a string we need to pass it as an array of string.
+    if (destName === 'End') {
+      destName = ['E', 'n', 'd'];
+    }
```

Comment:
What happens when the destName is null and we want to insert it at the end? Also, is this used here to concatenate the new string with the string already present?

Reply:
The following if condition only works when destaname is 'End' means we want to name the destination as 'End', if it is null then the if check will not works and it will not interfere with the code already written. In any case we want the destination to be named as "End" we need this check to make sure we pass End as a array of string and not a key name.

Reply:
I haven't noticed this at some other place when you were passing the keys value in setValue. I am not able to understand why it's used in this way over here.

Reply:
'End' is a official key (like Enter, Page up) which is present in webdriver protocol ptal [here](https://w3c.github.io/webdriver/#keyboard-actions). Hence when we pass 'End' like this the webdriverio thinks it is the official key that we are passing and not the string that we want to type so it presses the keyboard 'End' keys instead of typing 'End' in text box and hence the text box remains empty and test fails but when we pass it as ['E', 'n', 'd'] it actually presses E then n then d and we are able to type 'End' in the text box.

Reply:
I think then you need to update this comment above to sound more clearer.

Reply:
Done.